scripts/compile.ts

```diff
@@ -33,10 +33,13 @@ async function writeFactoryDeployerTransaction(contract: CompilerOutputContract,
 	const data = arrayFromHexString(deploymentBytecode)
 
 	if (!process.env.MNEMONIC) throw Error("MNEMONIC is required")
-	const signer = ethers.Wallet.fromMnemonic(process.env.MNEMONIC!!)
-	const signedEncodedTransaction = await signer.signTransaction({
-		nonce, gasPrice, gasLimit, value, data, chainId
-	})
+    const provider = new ethers.providers.JsonRpcProvider(process.env.RPC)
+    const signer = ethers.Wallet.fromMnemonic(process.env.MNEMONIC!!).connect(provider)
+    //if the network supports EIP-1559 transaction, use the EIP-1559 transaction type
+    const tx = await signer.populateTransaction({
+        nonce, gasPrice, gasLimit, value, data, chainId
```

Comment:
How are you using EIP-1559 here? You haven't specified the transaction type and you are not using the maxPriorityFeePerGas or maxBaseFeePerGas here, right

Reply:
I was also confused for a while, but the key here seems to be using the `.populateTransaction` method that adds all the necessary fields, including EIP-1559 gas fields. So the comment could be better 100%.

Also, possibly we'd need to support multiple transaction types because, like networks only supporting eip-1559 transactions, there are networks that only support legacy ones.

Reply:
I see, didn't know the populateTransaction method does that by default

Reply:
The function `sendTransaction` in ethers.js' abstract Signer is as follows:
```javascript
// Populates all fields in a transaction, signs it and sends it to the network
async sendTransaction(transaction: Deferrable<TransactionRequest>): Promise<TransactionResponse> {
    this._checkProvider("sendTransaction");
    const tx = await this.populateTransaction(transaction);
    const signedTx = await this.signTransaction(tx);
    return await this.provider.sendTransaction(signedTx);
}
```
Before signing the transaction, it calls the populateTransaction method to fill in the relevant transaction fields, including the identification of the transaction type. If the type field is not provided, it checks whether the current chain supports EIP-1559. If it does, it populates the maxFeePerGas and maxPriorityFeePerGas fields and generates an EIP-1559 transaction. Otherwise, it generates a Legacy Transaction.

The current adaptation is made to align with the default behavior of the ethers.js library in order to support all EVM-compatible chains.

Reply:
`sendTransaction` does not work with all clients, and therefore should not be static.  The main issue I have with doing any of these integrations is whether or not a joint library should be created to merge both ethers and ethers-quorum together.

Reply:
I apologize for missing my point. There are several methods of doing things to accomplish the goals in mind, and where I really was going with my comment was that the use of: `populateTransaction` should not be allowed if there are not corresponding calls in the other main clients.